fix(networks): enable network before mint or airdrop

jrainville opened the pull request:
Fixes #15507

Makes sure to enable the network where the owner token was minted before minting or airdroping a token. This makes sure that the account selector shows the right balance and there is no ETH error before doing the transation

[activate-network.webm](https://github.com/user-attachments/assets/49c60a4b-3f7d-4ae3-a431-8c74d438c434)

noeliaSD reviewed (commented):
Hey @jrainville !! I'm not completely sure about this fix but to be more accurate on my proposal I should dig into wallet.. Maybe you already know how to do that.. What I'm thinking is that, it's a bit strange to "automatically enable" a network from a view different than the wallet main view, if it affects the main wallet view. I agree the enabled networks in wallet should not affect the airdrop and minting operations that's why my though is that what it should happen is that when the operation is triggered, it must proceed although the network is "disabled". Can we parametrize the operation to ignore the "enabled networks" and proceed, instead of manually altering the enabled networks? Hope it makes sense!

endulab commented:
We can't quitely change wallet settings. It is not a good approach. If some chain is not enabled we should inform the user about it and maybe enable it via message box, or ask about enabling it manually.

jrainville commented:
@noeliaSD @endulab you are right, it was not super ncie to do stuff in the background.

I updated the PR to show a warning with a button instead. You can see it in this video. Let me know what you think: 

[enableOpt.webm](https://github.com/user-attachments/assets/9ba9cd68-f0af-4d0f-a485-069a253bee69)

endulab commented:
@jrainville should we have this button also when Burn and Remote destruct?

noeliaSD commented:
> @noeliaSD @endulab you are right, it was not super ncie to do stuff in the background.
> 
> I updated the PR to show a warning with a button instead. You can see it in this video. Let me know what you think:
> 
>  enableOpt.webm

Have you consulted with the design team? I would suggest, instead of having a _button_, having a **switch** like in the rest of the cases... it seems a little cleaner to me from a UX point of view:

<img width="793" alt="Screenshot 2024-07-18 at 13 15 38" src="https://github.com/user-attachments/assets/ff3c1572-aefd-497e-a549-961db080354a">

And for the texts, I'd say something similar in both cases:
- `The owner token is minted on a network disabled in the Wallet. Enable %network`
- `The token selected is on a network disabled in the Wallet. Enable %network`

noeliaSD commented on ui/app/AppLayouts/Communities/views/EditAirdropView.qml:
```diff
@@ -45,6 +45,10 @@ StatusScrollView {
     required property bool feesAvailable
 
+    property string enabledChainIds
+    property string networkThatIsNotActive
```
It would be better to make it "private" since it's an internal implementation property.

noeliaSD commented on ui/app/AppLayouts/Communities/views/EditAirdropView.qml:
```diff
@@ -45,6 +45,10 @@ StatusScrollView {
 
+    property string enabledChainIds
+    property string networkThatIsNotActive
+    property int networkIdThatIsNotActive
```
Same here. It's an internal implementation detail, no need to be part of the component api.

noeliaSD commented on ui/app/AppLayouts/Communities/views/EditAirdropView.qml:
```diff
@@ -292,12 +299,25 @@ StatusScrollView {
                                 root.selectedHoldingsModel, ["key", "amount"])
                 }
 
+                function checkIfWeShouldShowNetworkWarning(entry) {
```
The function itself only checks if the given chain is part of `enabledChainIds`! ;)
```suggestion
                function isEnabledChain(chain) {
```

noeliaSD commented on ui/app/AppLayouts/Communities/views/EditAirdropView.qml:
```diff
@@ -292,12 +299,25 @@ StatusScrollView {
 
+                function checkIfWeShouldShowNetworkWarning(entry) {
+                    // If the tokens' network is not activated, show a warning to the user
+                    if (!root.enabledChainIds.includes(entry.networkId)) {
```
It would be nice to do an `if(entry)` at the very beginning

noeliaSD commented on ui/app/AppLayouts/Communities/views/EditCommunityTokenView.qml:
```diff
@@ -44,6 +44,9 @@ StatusScrollView {
     property bool isFeeLoading
 
+    property string networkThatIsNotActive
+    signal enableNetwork
```
No need of _chain ID_ here?

noeliaSD commented on ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml:
```diff
@@ -288,7 +293,7 @@ StackView {
                 chainId: newTokenPage.ownerTokenChainId
                 chainName: newTokenPage.chainName
-                chainIcon: newTokenPage.chainIcon
+                chainIcon: newTokenPage.chainIconownerTokenChainId
```
Typo???
```suggestion
                chainIcon: newTokenPage.chainIconOwnerTokenChainId
```

jrainville replied:
That's an oopise, I guess I pasted here by accident. Reverting

noeliaSD commented on ui/app/AppLayouts/Communities/panels/MintTokensSettingsPanel.qml:
```diff
@@ -49,6 +49,9 @@ StackView {
     property bool ownerOrTMasterTokenItemsExist: false
 
+    property string enabledChainIds
+    property string networkThatIsNotActive
```
If it's an internal implementation detail, please make it "private"

noeliaSD reviewed (commented):
Some minor comments added, otherwise, LGTM!

jrainville commented:
@noeliaSD @endulab thanks for the comments again.
I improved the PR again. I extracted the panel to a component and reused it everywhere.
I also added the panel to the burn and remote destruct.
Finally, I corrected the issues Noelia found.

[burnAndDestruct.webm](https://github.com/user-attachments/assets/c6200dda-edf3-4270-9666-2ad720ad9e0d)

jrainville commented:
@noeliaSD @endulab can I get a (final) review please :pray:

noeliaSD commented on ui/app/AppLayouts/Communities/panels/NetworkWarningPanel.qml:
```diff
@@ -0,0 +1,26 @@
+RowLayout {
+    id: root
+
+    property string networkThatIsNotActive
```
```suggestion
    property string networkName
```

noeliaSD reviewed (approved):
I didn't test it since I don't have the needed setup now. 

Code LGTM!
